src/audio/google_rtc_audio_processing.c

```diff
@@ -275,6 +402,10 @@ static struct comp_dev *google_rtc_audio_processing_create(
 	if (cd) {
 		rfree(cd->output_buffer);
 		rfree(cd->aec_reference_buffer);
+		if (cd->memory_buffer) {
```

Comment:
not sure about your API call but memory buffer does not need to be in the if as rfree(NULL) is a no-op